autogpt_platform/backend/backend/server/v2/store/db.py

```diff
@@ -1193,6 +1338,182 @@ async def get_pending_submissions(
+
+    Args:
+        status: Filter by submission status (PENDING, APPROVED, REJECTED)
+        search_query: Search by name, description, or user email
+        page: Page number for pagination
+        page_size: Number of items per page
+
+    Returns:
+        StoreListingsWithVersionsResponse with listings and their versions
+    """
+    logger.debug(
+        f"Getting admin store listings with status={status}, search={search_query}, page={page}"
+    )
+
+    try:
+        # Build the where clause for StoreListing
+        where_dict = {}
+        where_dict["isDeleted"] = False
+
+        sanitized_query = sanitize_query(search_query)
+        if sanitized_query:
+            # Find users with matching email
+            matching_users = await prisma.models.User.prisma().find_many(
+                where={"email": {"contains": sanitized_query, "mode": "insensitive"}},
+            )
+
+            user_ids = [user.id for user in matching_users]
+
+            # Set up OR conditions
+            where_dict["OR"] = [
+                {"slug": {"contains": sanitized_query, "mode": "insensitive"}},
+                {
+                    "Versions": {
+                        "some": {
+                            "name": {"contains": sanitized_query, "mode": "insensitive"}
+                        }
+                    }
+                },
+                {
+                    "Versions": {
+                        "some": {
+                            "description": {
+                                "contains": sanitized_query,
+                                "mode": "insensitive",
+                            }
+                        }
+                    }
+                },
+                {
+                    "Versions": {
+                        "some": {
+                            "subHeading": {
+                                "contains": sanitized_query,
+                                "mode": "insensitive",
+                            }
+                        }
+                    }
+                },
+            ]
+
+            # Add user_id condition if any users matched
+            if user_ids:
+                where_dict["OR"].append({"owningUserId": {"in": user_ids}})
+
+        # Calculate pagination
+        skip = (page - 1) * page_size
+
+        # Create proper Prisma types for the query
+        where = prisma.types.StoreListingWhereInput(**where_dict)
+        include = prisma.types.StoreListingInclude(Versions=True, OwningUser=True)
+
+        # Query listings with their versions
+        listings = await prisma.models.StoreListing.prisma().find_many(
+            where=where,
+            skip=skip,
+            take=page_size,
+            include=include,
+            order=[{"createdAt": "desc"}],
+        )
+
+        # Get total count for pagination
+        total = await prisma.models.StoreListing.prisma().count(where=where)
+        total_pages = (total + page_size - 1) // page_size
```

Comment:
Maybe this is fine since it's an admin dashboard.

Buy It's always good to avoid pagination, but you will always end up needing to scan the whole thing just to paginate. Where you can just do infinite scrolling/paging by using the indexed `createdAt`.